sources/packages/backend/libs/utilities/src/date-utils.ts

```diff
@@ -379,3 +379,26 @@ export const isSameOrAfterDate = (
 ): boolean => {
   return dayjs(endDate).isSameOrAfter(startDate);
 };
+
+/**
+ * Get the last year of the current fiscal year for a given date.
+ * The fiscal year starts on April 1st of the current year
+ * and it ends on March 31st of the next year, which means
+ * it can be referred as 2025-26. This method will return the last
+ * year of the current fiscal year, which is 2026 in this case.
+ * @param date the date to be checked.
+ * @returns fiscal year, for instance, 2026 for 2025-26 fiscal year.
+ */
+export function getFiscalYear(date: Date): number {
+  const referenceDate = dayjs(date);
+  const referenceYear = referenceDate.year();
+  // Fiscal year inclusive start date.
+  // The fiscal year starts on April 1st of the current year.
+  const startDate = new Date(referenceYear, 3, 1); // April 1st
+  if (referenceDate.isSameOrAfter(startDate, "day")) {
```

Comment:
As a suggestion. we can simplify the logic this way. 

```ts
if (referenceDate.month() >= 3) {
    // Any date at or after April 1st of the current year is in
    // the next year reference fiscal year convention.
    return referenceYear + 1;
  }

```

Reply:
I am removing the dayjs also as below.
```ts
export function getFiscalYear(date: Date): number {
  const referenceYear = date.getFullYear();
  if (date.getMonth() >= 3) {
    // Any date at or after April 1st of the current year is in
    // the next year reference fiscal year convention.
    return referenceYear + 1;
  }
  return referenceYear;
}
```